ui/hooks/useTokenFiatAmount.js

```diff
@@ -22,24 +25,48 @@ import { isEqualCaseInsensitive } from '../../shared/modules/string-utils';
  * @param {boolean} [overrides.showFiat] - If truthy, ensures the fiat value is shown even if the showFiat value from state is falsey
  * @param {boolean} hideCurrencySymbol - Indicates whether the returned formatted amount should include the trailing currency symbol
  * @returns {string} The formatted token amount in the user's chosen fiat currency
+ * @param {string} [chainId] - The chain id
  */
 export function useTokenFiatAmount(
   tokenAddress,
   tokenAmount,
   tokenSymbol,
   overrides = {},
   hideCurrencySymbol,
+  chainId = null,
 ) {
+  const allMarketData = useSelector(getMarketData);
+
   const contractExchangeRates = useSelector(
     getTokenExchangeRates,
     shallowEqual,
   );
+
+  const contractMarketData = chainId
+    ? Object.entries(allMarketData[chainId]).reduce(
+        (acc, [address, marketData]) => {
+          acc[address] = marketData?.price ?? null;
+          return acc;
+        },
+        {},
+      )
+    : null;
+
+  const tokenMarketData = chainId ? contractMarketData : contractExchangeRates;
+
```

Comment:
Could you help me understand when chainId will be null?

Reply:
chainId is an optional parameter on the hook `useTokenFiatAmount` , if chainId is not passed we should fallback on current network

Reply:
Oh yes right cause the fct is also used by other components outside the scope of this feature; thnx